src/dynamo0p3.py

```diff
@@ -3053,6 +3421,83 @@ def operator(self, arg):
                                                      ndf_name_from, size]),
                                  intent=intent, entity_decls=[text]))
 
+    def cma_operator(self, arg):
+        ''' add the CMA operator arguments to the argument list '''
+        from f2pygen import DeclGen
+        # The CMA operator itself
+        self._arglist.append(arg.name)
+        # Associated scalar parameters
+        nrow = arg.name + "_nrow"
+        _local_args = [nrow]
+        if arg.function_space_to.orig_name != \
+           arg.function_space_from.orig_name:
+            # If the to- and from-spaces are the same then so are ncol and
```

Comment:
I'm not quite sure what you mean. However, I've re-written the comment so that it is worded correctly for its location. By default we pass both `nrow` and `ncol` to a kernel but if the to- and from-spaces are the same then we only pass `nrow`.